subsys/bluetooth/host/mesh/cfg.c

```diff
@@ -2916,6 +2916,11 @@ static void hb_publish(struct k_work *work)
 
 	hb_send(model);
 
+	if (cfg->hb_pub.count == 0) {
+		k_delayed_work_cancel(&cfg->hb_pub.timer);
```

Comment:
Good catch, but I don't think you need to call k_delayed_work_cancel since the delayed work is not auto-rescheduled when this function is called. I.e. simply returning from this function should be sufficient.